### ISSUE TITLE
Preserve ordering of constants rather than sorting them alphabetically

### ISSUE THREAD

**Issue by jcollins-g** [dart-lang/dartdoc#1550]
Transfer from [old dartdoc discussion](https://docs.google.com/document/d/1tndhthlM9jFls1kA5YvPj-LMiaNhp1uSAnRStpxB370/edit?ts=5a134f5e#):

"Frequently, specific parts of a class, file or library are part of a mental group. For example, in DateTime we can group JANUARY, FEBRUARY, etc. Furthermore, for some of them (like the months and weekdays) we want to keep them in a specific order."

**Comment by kevmoo** [comment]
This would be something to flag as an option.

In *most* cases, I'd argue for alphabetical sort – to ease scanning.

If a package author wants to opt-out, it should be explicit – perhaps via annotation?

**Comment by jcollins-g** [comment]
Agreed.  Categorization such as that proposed in #1353 might help with this too, to group related items together in the docs.

**Comment by sigurdm** [comment]
We could do some kind of annotation on the containing element:

```
/// @preserve_child_order
enum Months {
  january,
  february,
  // etc.
}
```

And for the entire library:

```
/// @preserve_child_order
library;

class First {}
class Second {}
class Third {}
class Fourth {}
/// ...
```
